File: game_of_life/mrugger/ruby/mediator.grid_writer.rb
require './game.types.rb'


class MediatorGridWriter

  private

  @grid
  @writer


  public

  def initialize(grid, writer)
    @grid = grid
    @writer = writer
  end


  # Writer interface

  def draw(start_x, start_y)
    output_size = @writer.begin_board
    for x in start_x..(start_x + output_size.x - 1)
      @writer.begin_row
      for y in start_y..(start_y + output_size.y - 1)
        @writer.write_cell(@grid.node_value(x, y))
      end
      @writer.end_row
    end
    @writer.end_board
  end

end

File: game_of_life/mrugger/java/MediatorGridWriter.java


class MediatorGridWriter
{
  private IGridable _grid;
  private IWriteable _writer;

  public MediatorGridWriter(IGridable grid, IWriteable writer)
  {
    _grid = grid;
    _writer = writer;
  }


  public void draw(Pair position)
  {
    Pair window_size = _writer.begin_board();
    for (int x = 0; x < window_size.x; x++)
    {
      _writer.begin_row();
      for (int y = 0; y < window_size.y; y++)
      {
        _writer.write_cell(_grid.node_value(position.x + x, position.y + y));
      }
      _writer.end_row();
    }
    _writer.end_board();
  }
}
